src/postgres-util/src/lib.rs

```diff
@@ -250,38 +250,44 @@ pub async fn drop_replication_slots(conn: &str, slots: &[String]) -> Result<(),
     let (client, connection) = tokio_postgres::connect(&conn, tls).await?;
     tokio::spawn(connection);
 
+    let replication_client = connect_replication(conn).await?;
```

Comment:
Once we don't allow invalid Postgres sources to be created we can remove the non-replication client.